### ISSUE TITLE
Isom muxer not terminating file correctly

### ISSUE THREAD

**Issue by iw-an** [membraneframework/membrane_mp4_plugin#57]
I've written a recording endpoint using membrane_rtc_engine that looks like this:
```
    children = %{
      {:track_receiver, track_id} => %TrackReceiver{
        track: Map.fetch!(state.tracks, track_id),
        initial_target_variant: :high
      },
      {:depayloader, track_id} => Membrane.RTC.Engine.Track.get_depayloader(Map.fetch!(state.tracks, track_id)),
      {:video_parser, track_id} => %Membrane.H264.FFmpeg.Parser{
        alignment: :au,
        attach_nalus?: true,
        framerate: {1, 30}
      },
      {:video_payloader, track_id} => %Membrane.MP4.Payloader.H264{
        parameters_in_band?: false
      },
      {:muxer, track_id} => %Membrane.MP4.Muxer.ISOM{
        fast_start: false
      },
      {:sink, track_id} => %Membrane.File.Sink{
        location: Path.join(state.output_dir_path, track_id <> ".mp4")
      }
    }
```

It's taking an webRTC H264 stream and saving an MP4. For some reason the moov atom is never present, even with fast start enabled. 

ffprobile outputs the following:
```
[NULL @ 0x148004a90] Opening '/Users/iwan/Documents/recordings/webrtctransceiver0.mp4' for reading
[file @ 0x148104190] Setting default whitelist 'file,crypto,data'
[mov,mp4,m4a,3gp,3g2,mj2 @ 0x148004a90] Format mov,mp4,m4a,3gp,3g2,mj2 probed with size=2048 and score=100
[mov,mp4,m4a,3gp,3g2,mj2 @ 0x148004a90] ISO: File Type Major Brand: isom
[mov,mp4,m4a,3gp,3g2,mj2 @ 0x148004a90] moov atom not found
[AVIOContext @ 0x1481042e0] Statistics: 32768 bytes read, 0 seeks
```

No matter what I seem to do I can't get a graceful termination from the muxer into the output file. I can end the stream by disconnecting the streaming client and have also been removing the registered endpoint too.

Any suggestions here? 
I'm using version `0.16.2` due to membrane_rtc_engine not being able to support higher currently.

**Comment by iw-an** [comment]
During testing yesterday I managed to get a single file that worked correctly and using https://github.com/ponchio/untrunc taking in the working file with any of the broken files, it successfully finds and inserts the correct metadata

**Comment by iw-an** [comment]
In `finalize_mp4` I've logged out the following too:
```
update_mdat_actions: [
  event: {:output,
   %Membrane.File.SeekEvent{position: {:bof, 32}, insert?: false}},
  buffer: {:output,
   %Membrane.Buffer{
     payload: <<0, 25, 241, 51>>,
     pts: nil,
     dts: nil,
     metadata: %{}
   }}
]
moov: <<0, 0, 16, 88, 109, 111, 111, 118, 0, 0, 0, 108, 109, 118, 104, 100, 0, 0, 0,
  0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 3, 232, 1, 9, 245, 240, 0, 1, 0, 0, 1, 0, 0,
  0, 0, 0, 0, 0, 0, 0, ...>>
```

So `handle_end_of_stream` is being called. Just that the data isn't being written to in the correct place?

**Comment by iw-an** [comment]
More research today. 
Left is broken file, right is correct output.
![image](https://user-images.githubusercontent.com/25753391/217539576-cf216bf7-f3ba-4a3e-8f34-3995785546cd.png)

The entirety of the final moov header output is missing however the data from outputted moov is valid. I converted the data into hex and matched against.

Curious as to why this is not entering the final file.

**Comment by iw-an** [comment]
I've validated that the mdat header is correct. Just trying to figure out why the moov never gets placed in the correct position

**Comment by iw-an** [comment]
Okay I have figured out what is going on. It seems there's a weird interaction between web_rtc_engine and this plugin / pipelines in general. When I terminate the source of the stream the file is saved correctly with the moov data being appended to the file. when I call web_rtc_engine's `remove_endpoint` the data is never added.

I have debugged already and the moov data is returned from `handle_end_of_stream` from both methods is there.
